monitoring/generator.go

```diff
@@ -249,26 +249,45 @@ func (o Observable) validate() error {
 	return nil
 }
 
-// Alert defines when an alert would be considered firing.
-type Alert struct {
+func Alert() alert {
+	return alert{}
+}
+
+// alert defines when an alert would be considered firing.
+type alert struct {
```

Comment:
Maybe a name that is a bit more distinctive from the `Alert()` constructor?

```suggestion
// alertDefinition defines when an alert would be considered firing.
type alertDefinition struct {
```